packages/client/src/generation/TSClient/common.ts

```diff
@@ -93,15 +115,41 @@ In case this error is unexpected for you, please report it in https://github.com
 }
 
 export const commonCodeTS = ({ runtimeDir, runtimeName, clientVersion, engineVersion }: TSClientOptions) => ({
-  tsWithoutNamespace: () => `import * as runtime from '${runtimeDir}/${runtimeName}';
+  tsWithoutNamespace: (edge = false) => `${
+    // Mock out the runtime types for edge client
+    edge
+      ? `declare namespace runtime {
```

Comment:
Note

Uses mock namespace for `runtime` in `edge.d.ts`. The original type file (`runtime/index.d.ts`) has `@types/node` dependency and that doesn't work for Deno (I believe that's also something inconvenient for cloudflare workers as they don't have node APIs. So I think using mock here makes sense for both cf workers and deno)

While `runtime` types are mock now, `PrismaClient` and user defined data models still have full typings. So I believe this change doesn't cause any major negative impact to the developer experience.

Reply:
(https://github.com/prisma/prisma/pull/14386/files#r930355689)